test/smoke/framework/testCore/src/main/java/com/microsoft/applicationinsights/smoketest/docker/AiDockerClient.java

```diff
@@ -184,4 +188,21 @@ public boolean isContainerRunning(String id) throws IOException, InterruptedExce
 			sw.close();
 		}
 	}
+
+	public String createNetwork(String name) throws IOException, InterruptedException {
+	    Process p = buildProcess(dockerExePath, "network", "create", "--driver", "bridge", name).start();
+	    waitAndCheckCodeForProcess(p, 10, TimeUnit.SECONDS, "creating network");
+	    return getFirstLineOfProcessOutput(p);
+    }
+
+	private static String getFirstLineOfProcessOutput(Process p) throws IOException {
+		List<String> lines = CharStreams.readLines(new InputStreamReader(p.getInputStream()));
+		return lines.get(0);

```

Comment:
This is not needed. You are essentially reading the whole stream to just get the first line. It can be expensive operation. Instead just use, readFirstLine() method of CharStreams.

https://google.github.io/guava/releases/15.0/api/docs/com/google/common/io/CharStreams.html#readFirstLine(com.google.common.io.InputSupplier)

Reply:
That particular method was removed in guava 18.0. smoketests use version 20.0

Reply:
If you have reservations on using Guava for this purpose, I would not object you because it's test code and has less direct impact on customers. However, if you wish to you can use the Java's native method from BufferedReader to read the first line.

https://docs.oracle.com/javase/7/docs/api/java/io/BufferedReader.html#readLine()

Reply:
In the cases where this is used, there is always one line of output. The overhead from this function will be minimal, but I'll keep the methods you mentioned in mind. Thanks for the tips

Reply:
np :)